feat/chat-keyboard-1: 채팅 키보드 겹침 해결

KooSuYeon opened the pull request:
### 개요 

- ChatInputSend를 KeyboardAvoidingView로 겹치는 것이 아닌 ChatRoomPage에서KeyboardAvoidingView로 빼서 키보드%채팅 겹침 해결
- 친구 목록에서 JWT 채팅방 접근 가능하도록
- 채팅을 친 후에 자동으로 스크롤 되도록 handleScroll 추가
- 스크롤 바 추가
- 채팅 상단 툴바 marginBottom 조절  30 -> 10

---
테스트 화면

https://github.com/user-attachments/assets/9b670716-466d-40f8-95d4-137a82aa27fe 

+시간& 배터리 흰색으로 안뜨는 것 확인했습니다!
<img src="https://github.com/user-attachments/assets/3a3338ce-7766-43b3-baf6-6c4009135e9b" width="200" />

KooSuYeon committed:
feat/chat-keyboard-1: 채팅 키보드 겹침 해결

- ChatInputSend를 KeyboardAvoidingView로 겹치는 것이 아닌 ChatRoomPage에서KeyboardAvoidingView로 빼서 키보드%채팅 겹침 해결
- 채팅을 친 후에 자동으로 스크롤 되도록 handleScroll 추가
- 스크롤 바 추가

KooSuYeon committed:
feat/chat-keyboard-2: Lint Test

KooSuYeon committed:
feat/chat-keyboard-3:npm install --legacy-peer-deps

KooSuYeon committed:
feat/chat-keyboard-4: punyCode 추가 및 react-native-safe-area-context 4.7.1 버전 change

seungholee-dev commented on package.json:
```diff
@@ -67,14 +67,17 @@
 		"react-native-drawer-layout": "^3.3.0",
 		"react-native-fs": "^2.20.0",
 		"react-native-gesture-handler": "~2.16.1",
+		"react-native-gifted-chat": "^2.6.4",
+		"react-native-input-accessory-view": "^0.4.0",
+		"react-native-keyboard-aware-scroll-view": "^0.9.5",
```
혹시 해당 패키지는 어떤 용도로 추가되었을까요?!
사용되는 부분이 보이지 않아서요!

seungholee-dev commented on package.json:
```diff
@@ -57,6 +57,8 @@
 		"expo-status-bar": "~1.12.1",
 		"expo-updates": "^0.25.25",
 		"i18next": "^23.14.0",
+		"install": "^0.13.0",
+		"npm": "^10.9.0",
```
이 부분은 불필요할 것 같아서 삭제해도 괜찮을 것 같네요!

seungholee-dev reviewed (commented):
한번 이 두 부분 삭제 후에 다시 진행해보시겠어요?

KooSuYeon committed:
feat/chat-keyboard-5: package.json 불필요사항 삭제

KooSuYeon committed:
feat/chat-keyboard-6: 채팅방 친구 목록에서 채팅방 접근 JWT 부여

- 빠트렸던 채팅방 친구 목록에서 채팅방 접근 시에 JWT가 부여될 수 있도록 수정
- ChatroomPage에서도 JWT 부여
- 채팅방 상단 툴 바 marginBottom 줄이기

KooSuYeon committed:
feat/chat-keyboard-7: eLint test

KooSuYeon commented:
@seungholee-dev 
채팅할 때에 키보드 겹침 해결 & 최신 채팅 자동 스크롤 & 친구목록에서 JWT 채팅방 접근 가능하도록 한 PR dev에 merge하겠습니다
